### PR TITLE
Enable apps/v1beta2 in e2e test clusters

### PR DESCRIPTION
**Pull request by janetkuo**
We recently added apps/v1beta2, but we cannot enable it by default because it's still WIP. We will enable apps/v1beta2 by default in 1.8 release. For now, force enabling it in all e2e test clusters so that we can add e2e tests for it. 

cc @fejta

### CODE REVIEW

File: kubetest/e2e.go [hunk 1]
```diff
@@ -37,6 +37,9 @@ func run(deploy deployer, o options) error {
 	os.Setenv("KUBE_CONFIG_FILE", "config-test.sh")
 	// force having batch/v2alpha1 always on for e2e tests
 	os.Setenv("KUBE_RUNTIME_CONFIG", "batch/v2alpha1=true")
+	// force having apps/v1beta2 always on for e2e tests
+	// TODO: remove it once we enable apps/v1beta2 by default
+	os.Setenv("KUBE_RUNTIME_CONFIG", "apps/v1beta2=true")
```
Comment (fejta) [inline kubetest/e2e.go]: a) looks like this invalidates L39
b) how is this used? New code should follow the pop := pushEnv(); defer pop() pattern used elsewhere to explicitly set this where it is used and revert the change after the command that needs it is finished.

Comment (janetkuo) [inline kubetest/e2e.go]: I cannot find `test_context.go` in this repo. Where's that file and what flag be set?

Is it okay if I instead change L39 into:

```go 
os.Setenv("KUBE_RUNTIME_CONFIG", "batch/v2alpha1=true,apps/v1beta2=true")
```
and then revert this change once apps/v1beta2 is enabled by default? We only need this temporarily (during 1.8 implementation stage) so that we can use this new version in e2e.

Comment (crimsonfaith91) [inline kubetest/e2e.go]: Will this work: `export KUBE_RUNTIME_CONFIG="batch/v2alpha1=true,apps/v1beta2=true"`?

Comment (crimsonfaith91) [inline kubetest/e2e.go]: I think it should be able to work by implying from this [comment](https://github.com/kubernetes/kubernetes/issues/16117#issue-112872599). :)

Comment (fejta) [inline kubetest/e2e.go]: test context is here: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/test_context.go

How is this environment variable used?

We are trying very hard to eliminate implicit dependencies on global variables. We should identify which programs kubetest calls need this config and explicitly send it to those programs.

Also, we should be able to modify this value without shipping a new kubekins image. Will you convert this into a kubetest flag? https://github.com/kubernetes/test-infra/blob/master/kubetest/main.go#L92 

Is this setting really the right one for every job? Including ones from say 1.5? If not then let's update `jobs/config.json` to add this `--new-flag=batch/v2alpha1=true,apps/v1beta2=true` to the relevant jobs

Comment (crimsonfaith91) [inline kubetest/e2e.go]: @fejta Referring [this](https://github.com/kubernetes/kubernetes/blob/master/cluster/vagrant/config-default.sh#L96-L97),  the environment variable triggers behaviors of inputted version.

Currently, `v1beta2` is disabled by default. We need to turn on `v1beta2` behaviors for testing purpose during implementation stage.

**Comment by janetkuo** [comment]
@fejta I believe I've addressed your comments by adding a new flag `--runtime-config` for this env.

File: kubetest/e2e.go (hunk 1 above)
Comment (janetkuo) [inline kubetest/e2e.go]: > How is this environment variable used?

@fejta This env is eventually converted to `--runtime-config` passed to api-server to enable/disable API versions.

**Comment by janetkuo** [comment]
@fejta I'm not familiar with test infra codebase. How is this new kubetest flag consumed? How do I specify it in my e2e tests?

File: kubetest/e2e.go [hunk 2]
```diff
@@ -35,8 +35,13 @@ func run(deploy deployer, o options) error {
...
 	}
 	os.Setenv("KUBE_CONFIG_FILE", "config-test.sh")
+
 	// force having batch/v2alpha1 always on for e2e tests
-	os.Setenv("KUBE_RUNTIME_CONFIG", "batch/v2alpha1=true")
+	runtimeConfig := "batch/v2alpha1=true"
+	if o.runtimeConfig != "" {
+		runtimeConfig = runtimeConfig + "," + o.runtimeConfig
+	}
+	os.Setenv("KUBE_RUNTIME_CONFIG", runtimeConfig)
```
Comment (fejta) [inline kubetest/e2e.go]: I would prefer we just set the var to the flag's value if it's non-empty and then default the flag value to batch/v2alpha1=true

Comment (janetkuo) [inline kubetest/e2e.go]: Whatever I set to this flag would invalidate the default value. Would that be okay?

Comment (fejta) [inline kubetest/e2e.go]: Yes

**Comment by fejta** [comment]
First we'll need to release a new kubetest binary that has this new flag.

Then the job in jobs/config.json where you want to change the flag's value and add set the flag there: https://github.com/kubernetes/test-infra/blob/master/jobs/config.json#L9453

File: kubetest/e2e.go (hunk 2 above)
Comment (janetkuo) [inline kubetest/e2e.go]: Done

**Comment by fejta** [comment]
Feel free to click the merge button when you're ready (no submit queue yet) and/or let me know and I can do so

**Comment by janetkuo** [comment]
@fejta please merge it. I don't have write permission :P

**Comment by fejta** [comment]
Done :) will send you a PR that updates our test images with the new version later tonight/tomorrow

**Comment by janetkuo** [comment]
@fejta any update on this?

**Comment by krzyzacy** [comment]
This is already deployed, you can add `--runtime-config=` in https://github.com/kubernetes/test-infra/blob/master/jobs/config.json for the job(s) you are interested in.

**Comment by crimsonfaith91** [comment]
If I want to add the flag for ci-kubernetes-e2e-gce-etcd3 test, how many jobs will be affected.

**Comment by krzyzacy** [comment]
One job has one entry - so only that job will be affected :-) And I completely forgot this PR already went in.

**Comment by crimsonfaith91** [comment]
@foxish
